core/components/mediamanager/model/mediamanager/classes/files.class.php

```diff
@@ -1964,6 +2047,12 @@ public function archiveFiles($selectedFiles)
      */
     public function archiveReplaceFile($fileId, $newFileId)
     {
+        if ($fileId === $newFileId) {
+            return [
+                'message' => $this->mediaManager->modx->lexicon('mediamanager.files.archive_and_replace.success')
```

Comment:
Not sure silently claiming success is the right approach. Archiving and replacing a file by itself should not be possible anyway.